packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -85,71 +91,63 @@ async function proxyPreviewModeApiRouteHandler(
+     ...headers
+  } = req.headers
 
-    proxyReq.path = url.pathname + url.search
-  })
-
-  if (req.query.secret !== apiKey) return res.status(401).send('Unauthorized')
+  const isForwardedProtoHttps = match(req.headers['X-Forwarded-Proto'])
+    .with(P.string, x => x.split(','))
+    .with(P.array(P.string), x => x)
+    .otherwise(() => [])
+    .includes('https')
 
-  const host = req.headers.host
-  const originalCookies = req.headers.cookie
+  const isForwardedSSL = match(req.headers['X-Forwarded-SSL'])
+    .with('on', () => true)
+    .otherwise(() => false)
 
-  if (host == null) return res.status(400).send('Bad Request')
-
-  const forwardedProtoHeader = req.headers['x-forwarded-proto']
-
-  let forwardedProto: string[] = []
-  if (Array.isArray(forwardedProtoHeader)) {
-    forwardedProto = forwardedProtoHeader
-  } else if (typeof forwardedProtoHeader === 'string') {
-    forwardedProto = forwardedProtoHeader.split(',')
-  }
-
-  const isForwardedProtoHttps = forwardedProto.includes('https')
+  const proto = isForwardedProtoHttps || isForwardedSSL ? 'https' : 'http'
+  const url = new URL(`${proto}://${req.headers.host}${req.url}`)
 
-  const forwardedSSL = req.headers['x-forwarded-ssl']
-  const isForwardedSSL = typeof forwardedSSL === 'string' && forwardedSSL === 'on'
+  url.searchParams.delete('x-makeswift-preview-mode')
 
-  const proto = isForwardedProtoHttps || isForwardedSSL ? 'https' : 'http'
-  let target = `${proto}://${host}`
-
-  // If the user generates a locally-trusted CA for their SSL cert, it's likely that Node.js won't
-  // trust this CA unless they used the `NODE_EXTRA_CA_CERTS` option
-  // (see https://stackoverflow.com/a/68135600). To provide a better developer experience, instead
-  // of requiring the user to provide the CA to Node.js, we don't enforce SSL during local
-  // development.
-  const secure = process.env['NODE_ENV'] === 'production'
-
-  const previewData: MakeswiftPreviewData = {
-    makeswift: true,
-    siteVersion: MakeswiftSiteVersion.Working,
-  }
   const setCookie = res.setPreviewData(previewData).getHeader('Set-Cookie')
-  res.removeHeader('Set-Cookie')
 
   if (!Array.isArray(setCookie)) return res.status(500).send('Internal Server Error')
 
   const additionalCookies = parse(setCookie)
     .map(cookie => serialize(cookie.name, cookie.value, cookie as CookieSerializeOptions))
     .join(';')
 
+  const originalCookies = req.headers.cookie
   const cookie = originalCookies ? `${originalCookies}; ${additionalCookies}` : additionalCookies
 
-  return await new Promise<void>((resolve, reject) =>
-    previewModeProxy.web(req, res, { target, headers: { cookie }, secure }, err => {
-      if (err) reject(err)
-      else resolve()
-    }),
-  )
+  const response = await fetch(url, {
+    referrer: req.headers.referer,
+    headers: {
+      ...headers as Record<string, string>,
+      'Set-Cookie': cookie,
+    },
+  });
+
+  res.removeHeader('Set-Cookie')
```

Comment:
How come we're removing the `Set-Cookie` header from the response?

Reply:
Oh, I see now that this was to remove the Preview Mode cookie. We should keep this colocated with the `res.setPreviewData` call.